Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -2746,14 +2754,15 @@ static void setPlaybackFlags(GstElement* pipeline, bool isMediaStream)
 void MediaPlayerPrivateGStreamer::createGSTPlayBin(const URL& url)
 {
     GST_INFO("Creating pipeline for %s player", m_player->isVideoPlayer() ? "video" : "audio");
-    const char* playbinName = "playbin";
 
-    // MSE and Mediastream require playbin3. Regular playback can use playbin3 on-demand with the
-    // WEBKIT_GST_USE_PLAYBIN3 environment variable.
-    const char* usePlaybin3 = g_getenv("WEBKIT_GST_USE_PLAYBIN3");
+    // MSE and Mediastream require playbin3. Regular playback uses playbin3 if GStreamer >= 1.20 is
+    // detected, unless the WEBKIT_GST_USE_PLAYBIN2 environment variable is set to 1.
+    const char* playbinName = "playbin3";
+    auto usePlaybin2Override = StringView::fromLatin1(g_getenv("WEBKIT_GST_USE_PLAYBIN2"));
+    auto usePlaybin2 = usePlaybin2Override ? parseInteger<unsigned>(usePlaybin2Override) : 0;
     bool isMediaStream = url.protocolIs("mediastream"_s);
-    if (isMediaSource() || isMediaStream || (usePlaybin3 && !strcmp(usePlaybin3, "1")))
-        playbinName = "playbin3";
+    if ((*usePlaybin2 || !webkitGstCheckVersion(1, 20, 0)) && !isMediaSource() && !isMediaStream)
```

Comment:
I guess you should be checking if `usePlaybin2` `has_value` or use `value_or`.

Reply:
True! I need to check test failures too.